[WIP] Update for the newest palserver version (v0.6.2)

Lulalaby opened the pull request:
### Description

- Added support for the new config values and removed old ones
- Bumped packages

### TODOs

- [ ] #25
- [ ] #29
- [ ] #35
- [X] #40 
- [ ] #42

### Preview

<details>
<summary>Config Screenshots</summary>

![Settings Part 1](https://github.com/user-attachments/assets/850b0803-eed6-4b64-afc2-4c7745925448)
![Settings Part 2](https://github.com/user-attachments/assets/4ea3ea4c-f889-4eb5-ba4b-9470c20821cc)

</details>

### References

#### Configurations

<details>
<summary>Default Configuration</summary>

```ini
; This configuration file is a sample of the default server settings.
; Changes to this file will NOT be reflected on the server.
; To change the server settings, modify Pal/Saved/Config/WindowsServer/PalWorldSettings.ini.
[/Script/Pal.PalGameWorldSettings]
OptionSettings=(
   Difficulty=None,
   RandomizerType=None,
   RandomizerSeed="",
   bIsRandomizerPalLevelRandom=False,
   DayTimeSpeedRate=1.000000,
   NightTimeSpeedRate=1.000000,
   ExpRate=1.000000,
   PalCaptureRate=1.000000,
   PalSpawnNumRate=1.000000,
   PalDamageRateAttack=1.000000,
   PalDamageRateDefense=1.000000,
   PlayerDamageRateAttack=1.000000,
   PlayerDamageRateDefense=1.000000,
   PlayerStomachDecreaceRate=1.000000,
   PlayerStaminaDecreaceRate=1.000000,
   PlayerAutoHPRegeneRate=1.000000,
   PlayerAutoHpRegeneRateInSleep=1.000000,
   PalStomachDecreaceRate=1.000000,
   PalStaminaDecreaceRate=1.000000,
   PalAutoHPRegeneRate=1.000000,
   PalAutoHpRegeneRateInSleep=1.000000,
   BuildObjectHpRate=1.000000,
   BuildObjectDamageRate=1.000000,
   BuildObjectDeteriorationDamageRate=1.000000,
   CollectionDropRate=1.000000,
   CollectionObjectHpRate=1.000000,
   CollectionObjectRespawnSpeedRate=1.000000,
   EnemyDropItemRate=1.000000,
   DeathPenalty=All,
   bEnablePlayerToPlayerDamage=False,
   bEnableFriendlyFire=False,
   bEnableInvaderEnemy=True,
   bActiveUNKO=False,
   bEnableAimAssistPad=True,
   bEnableAimAssistKeyboard=False,
   DropItemMaxNum=3000,
   DropItemMaxNum_UNKO=100,
   BaseCampMaxNum=128,
   BaseCampWorkerMaxNum=15,
   DropItemAliveMaxHours=1.000000,
   bAutoResetGuildNoOnlinePlayers=False,
   AutoResetGuildTimeNoOnlinePlayers=72.000000,
   GuildPlayerMaxNum=20,
   BaseCampMaxNumInGuild=4,
   PalEggDefaultHatchingTime=72.000000,
   WorkSpeedRate=1.000000,
   AutoSaveSpan=30.000000,
   bIsMultiplay=False,
   bIsPvP=False,
   bHardcore=False,
   bPalLost=False,
   bCharacterRecreateInHardcore=False,
   bCanPickupOtherGuildDeathPenaltyDrop=False,
   bEnableNonLoginPenalty=True,
   bEnableFastTravel=True,
   bIsStartLocationSelectByMap=True,
   bExistPlayerAfterLogout=False,
   bEnableDefenseOtherGuildPlayer=False,
   bInvisibleOtherGuildBaseCampAreaFX=False,
   bBuildAreaLimit=False,
   ItemWeightRate=1.000000,
   CoopPlayerMaxNum=4,
   ServerPlayerMaxNum=32,
   ServerName="Default Palworld Server",
   ServerDescription="",
   AdminPassword="",
   ServerPassword="",
   PublicPort=8211,
   PublicIP="",
   RCONEnabled=False,
   RCONPort=25575,
   Region="",
   bUseAuth=True,
   BanListURL="https://api.palworldgame.com/api/banlist.txt",
   RESTAPIEnabled=False,
   RESTAPIPort=8212,
   bShowPlayerList=False,
   ChatPostLimitPerMinute=30,
   CrossplayPlatforms=(Steam,Xbox,PS5,Mac),
   bIsUseBackupSaveData=True,
   LogFormatType=Text,
   SupplyDropSpan=180,
   EnablePredatorBossPal=True,
   MaxBuildingLimitNum=0,
   ServerReplicatePawnCullDistance=15000.000000,
   bAllowGlobalPalboxExport=True,
   bAllowGlobalPalboxImport=False,
   EquipmentDurabilityDamageRate=1.000000,
   ItemContainerForceMarkDirtyInterval=1.000000
)
```

</details>

<details>
<summary>Used Configuration</summary>

```ini
[/Script/Pal.PalGameWorldSettings]
OptionSettings=(
   Difficulty=None,
   RandomizerType=None,
   RandomizerSeed="",
   bIsRandomizerPalLevelRandom=False,
   DayTimeSpeedRate=1.0,
   NightTimeSpeedRate=1.2,
   ExpRate=1.5,
   PalCaptureRate=10.0,
   PalSpawnNumRate=1.0,
   PalDamageRateAttack=1.0,
   PalDamageRateDefense=1.0,
   PlayerDamageRateAttack=1.0,
   PlayerDamageRateDefense=1.0,
   PlayerStomachDecreaceRate=0.4,
   PlayerStaminaDecreaceRate=0.4,
   PlayerAutoHPRegeneRate=1.5,
   PlayerAutoHpRegeneRateInSleep=3.0,
   PalStomachDecreaceRate=0.3,
   PalStaminaDecreaceRate=0.3,
   PalAutoHPRegeneRate=1.5,
   PalAutoHpRegeneRateInSleep=3.0,
   BuildObjectHpRate=1.0,
   BuildObjectDamageRate=-1.0,
   BuildObjectDeteriorationDamageRate=-1.0,
   CollectionDropRate=5.0,
   CollectionObjectHpRate=0.8,
   CollectionObjectRespawnSpeedRate=1.0,
   EnemyDropItemRate=5.0,
   DeathPenalty=None,
   bEnablePlayerToPlayerDamage=False,
   bEnableFriendlyFire=True,
   bEnableInvaderEnemy=False,
   EnablePredatorBossPal=True,
   bActiveUNKO=False,
   bEnableAimAssistPad=True,
   bEnableAimAssistKeyboard=False,
   DropItemMaxNum=10000,
   DropItemMaxNum_UNKO=100,
   BaseCampMaxNum=128,
   BaseCampMaxNumInGuild=20,
   BaseCampWorkerMaxNum=48,
   DropItemAliveMaxHours=1.0,
   bAutoResetGuildNoOnlinePlayers=False,
   AutoResetGuildTimeNoOnlinePlayers=72.0,
   GuildPlayerMaxNum=20,
   PalEggDefaultHatchingTime=0.1,
   WorkSpeedRate=4.0,
   AutoSaveSpan=30.0,
   CrossplayPlatforms=(Steam,Xbox,PS5,Mac),
   LogFormatType=Text,
   bIsMultiplay=True,
   bIsPvP=False,
   bHardcore=False,
   bPalLost=False,
   bCharacterRecreateInHardcore=False,
   bCanPickupOtherGuildDeathPenaltyDrop=True,
   bEnableNonLoginPenalty=True,
   bEnableFastTravel=True,
   bIsStartLocationSelectByMap=True,
   bExistPlayerAfterLogout=True,
   bEnableDefenseOtherGuildPlayer=True,
   bInvisibleOtherGuildBaseCampAreaFX=False,
   bBuildAreaLimit=False,
   ItemWeightRate=0.001,
   bShowPlayerList=True,
   CoopPlayerMaxNum=4,
   ServerPlayerMaxNum=16,
   ServerName="<redacted>",
   ServerDescription="<redacted>",
   AdminPassword="<redacted>",
   ServerPassword="<redacted>",
   PublicPort=<redacted>,
   PublicIP="<redacted>",
   RCONEnabled=True,
   RCONPort=25575,
   RESTAPIEnabled=True,
   RESTAPIPort=<redacted>,
   bIsUseBackupSaveData=True,
   Region="EU",
   bUseAuth=True,
   BanListURL="https://api.palworldgame.com/api/banlist.txt",
   SupplyDropSpan=120,
   ChatPostLimitPerMinute=100,
   MaxBuildingLimitNum=0,
   ServerReplicatePawnCullDistance=15000.0,
   bAllowGlobalPalboxExport=True,
   bAllowGlobalPalboxImport=True,
   EquipmentDurabilityDamageRate=0.6,
   ItemContainerForceMarkDirtyInterval=1.0
)
```

</details>

#### Others

<details>
<summary>OpenAPI Specification</summary>

Taken from https://github.com/Lulalaby/palworld-openapi-spec

</details>

### GitHub Commands

Fixes #40
Closes #42

Lulalaby committed:
fix: update for palworld server v0.6.1.75453

fix: don't override additional configs

Lulalaby commented:
@diogomartino lmk if you don't want this, want something different or whatever.
I'm actively using it for a modded palworld atm, so I thought k... let's fix it

Lulalaby committed:
fix: add more readme instructions

Lulalaby committed:
fix: revert bumping version for now

Lulalaby committed:
temp fix since i was too tired to save the og config as state

Lulalaby commented:
@Seb1Zero if you still run a server and wanna give feedback, lmk :)

diogomartino commented:
@Lulalaby hey, thank you for your work. I had no idea people still used this, and since I don't play this anymore, I kinda forgot about it. I'm gonna check your changes in a bit.

diogomartino commented:
Looks good, I have a question that impacts nothing, just for curiosity. Why the `onPress` instead of `onClick`? Also, based on your screenshots the radio buttons look off, you should use the `next-ui` components to maintain the look and feel.

Lulalaby commented:
Yeee I'm playing around with better radio buttons later on.
onClick was throwing a deprecated warning after bumping packages.
Why react decided to name it onPress, no fucking clue. It's weird.
Seems to work the same way tho

diogomartino commented:
That onPress is really strange. Probably not react but the ui library or something. I might pull the code this weekend to take a look

Lulalaby commented:
![image](https://github.com/user-attachments/assets/68ae8fcf-b56f-4cc7-8c23-c294c3e20c62)
ye ig the ui library

Lulalaby committed:
fix: don't force rcon to use 127.0.0.1 if we connect remotely

This removes the hardcoded 127.0.0.1 value for rcon and sets the host on the init screen instead, later reading it for the connection

Lulalaby committed:
fix: sizes

Lulalaby committed:
fix: make radio great again!

Lulalaby commented:
Radio has been fixed
![image](https://github.com/user-attachments/assets/1d39024f-f901-42af-8761-8bf2f34f55d6)

Lulalaby commented:
In regards to #39: After some research it seems our current version of nextui does not handle it correctly and monkey patching isn't possible either.
tl;dr cuz value 0 is false, it goes empty
I _tried_ upgrading nextui (now heroui) but tailwindcss etc blew up, so I reverted it. I don't have the nerves to deal with that.

Lulalaby committed:
feat: open api schema (by me)

Create .hintrc
chore: cleanup server-config.ts and sort keys
Only exception is ConfigKey, since that decides how the frontend shows them

Lulalaby committed:
feat: openapi client generator

This may make things easier for us.
Saving this temporary, not tested

Lulalaby committed:
Revert "fix: don't force rcon to use 127.0.0.1 if we connect remotely"

This reverts commit 133f0e729f1045e20cffa566a285eddf09c17d16.

I forgot that calls go to the ds server and get executed there .-.

Lulalaby committed:
fix: client generation

diogomartino commented:
I'll take care of that stuff this weekend.

Lulalaby commented:
What would you prefer in regards to the REST API
Should the server handle the calls and the gui request infos through the websocket (would mean duplicate code tho)
Or should the gui do the API calls directly, and get the login data through the PalworldGameSettings.ini (which would mean tho if it's a remote / web app connections, it might not work, unless the users publish the port, which would mean other security flaws)

Lulalaby committed:
fix(experimental): Do not kill a server process if it does not belong to us

THis aims to check if the exec path is actually within the palworld gui dir

diogomartino commented:
Damn, I didn't even know the server now has a REST API, if I remember correctly I was getting the player list and such via rcon. That changes things quite a lot because most features the app has can be deprecated. I would say that calling the API directly from the frontend is the best approach here because there are features that can be used without having GUI server running along (like the admin panel where you see the players list).

So I would say that to have all the features available you would still need to download the GUI server, but any feature supported by the API should be open for everyone to use. Lmk what you think.

Lulalaby commented:
Sounds reasonable enough to expose it on the frontend. We just need to add a note or warning for web users then, that they might need to expose the port if they're not calling from localhost.
Tho that might be a bit tricky again. ref #29

NookieCrusher commented:
Hi, not sure where to put this so sorry if in wrong place. How do I host 2 servers using your GUI. Keeps launching on 8211 but need a second on 8212. Even after changing the ini file ports.

Also missing some options that are new in the GUI such as weapon/tool degradation, item weights etc. Also breaks the ini file if applied manually then update on the GUI. Thanks

Lulalaby commented:
@NookieCrusher Please do not comment on pull requests
Furthermore, your issue is #40 
But that's fixed with this PR

Also I'm gonna add support to allow more instances

Lulalaby committed:
fix: rcon kickplayer & banplayer is kick & ban now

Lulalaby committed:
fix(temp): allow copying playerid to clipboard on admin

To be completely honest, we shouldn't do this at all anymore.
Palworld isn't steam-only anymore. It's also crossplatform now.
So we get GTK, Steam and other ids for players.
The current implementation is faulty.
Will address these issues later

Lulalaby commented:
Fyi taking a break of a few days before continuing. Need to recharge my "battery"

diogomartino commented:
Go ahead :muscle:

Lulalaby commented:
Kinda forgot about that lmao. Star Citizen got the better of me. Gonna check the next days

Lulalaby commented:
Unfortunately I got really busy with work the past months and I do not run my own palworld server anymore because everyone lost interest. We can either merge this as is, or scrape it. Idm either
